extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -15,4 +32,37 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols() {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true, GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void registerVertxGraphqlUI(VertxGraphqlRecorder recorder,
+            BuildProducer<NativeImagePackageResourceBuildItem> nativeResourcesProducer, VertxGraphqlConfig config,
+            LaunchModeBuildItem launchMode, BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints,
+            BuildProducer<RouteBuildItem> routes) {
+        boolean includeVertxGraphqlUi = launchMode.getLaunchMode().isDevOrTest() || config.ui.alwaysInclude;
+
+        if (!includeVertxGraphqlUi) {
+            return;
+        }
+
+        String path = config.ui.path;
+        if ("/".equals(path)) {
+            throw new ConfigurationError(
+                    "quarkus.vertx-graphql.ui.path was set to \"/\", this is not allowed as it blocks the application from serving anything else.");
+        }
+
+        Handler<RoutingContext> handler = recorder.handler(path);
```

Comment:
I think you should probably remove a potential trailing slash from the value.

Reply:
Good catch.

Reply:
done